src/h3lib/lib/h3UniEdge.c

```diff
@@ -229,14 +242,25 @@ void H3_EXPORT(getH3UnidirectionalEdgeBoundary)(H3Index edge, GeoBoundary* gb) {
 
     int k = 0;
     for (int i = 0; i < origin.numVerts; i++) {
-        for (int j = 0; j < destination.numVerts; j++) {
-            if (geoAlmostEqualThreshold(&origin.verts[i], &destination.verts[j],
-                                        0.000001)) {
-                gb->verts[k].lat = origin.verts[i].lat;
-                gb->verts[k].lon = origin.verts[i].lon;
+        if (_hasMatchingVertex(&origin.verts[i], &destination)) {
+            // If we are on vertex 0, we need to handle the case where it's the
+            // end of the edge, not the beginning.
+            // TODO: This is fine for this limited case, but it might be
+            // better/cleaner to use a VertexGraph for this
+            if (i == 0 &&
+                !_hasMatchingVertex(&origin.verts[i + 1], &destination)) {
+                postponedVertex = origin.verts[i];
+                hasPostponedVertex = true;
+            } else {
+                gb->verts[k] = origin.verts[i];
                 k++;
             }
         }
     }
+    // If we postponed adding the last vertex, add it now
+    if (hasPostponedVertex) {
+        gb->verts[k] = postponedVertex;
+        k++;
+    }
     gb->numVerts = k;
-}
+}
```

Comment:
Deleted newline from end of file. Put a little marker here to show me.

Reply:
Hm. I'm surprised `clang-format` doesn't have some sort of opinion on that. Will fix.